Framework/DataHandling/src/GroupDetectors2.cpp

```diff
@@ -669,29 +670,25 @@ void GroupDetectors2::processMatrixWorkspace(
 
     // translate to detectors
     std::vector<detid_t> det_ids;
-    if (spectrumInfo.hasDetectors(i)) {
+
+    // If the detector was not found or was not in a group, then ignore it.
+    if (spectrumInfo.spectrumDefinition(i).size() > 1) {
```

Comment:
Good. I think you can now remove the `DetectorGroup` from the includes for this algorithm.